data/opam_docs/2.1/03_faq.md

```diff
@@ -0,0 +1,446 @@
+  `~/.opam` and get back to `opam init`
+
+---
+
+#### 🐫  What are the minimum requirements?
+
+1GB of memory should be all you need. It was reported that you may run into
+problems with 512MB of RAM and no swap. Of course, compiling the packages may
+need more.
+
+---
+
+#### 🐫  Some packages fail during compilation, complaining about missing dependencies ("m4", "libgtk", etc.)
+
+> NOTE: since opam 2.1.0, the following is directly handled by opam, without
+> relying on a plugin.
+
+They probably depend on system, non-OCaml libraries: they need to be installed
+using your system package manager (apt-get, yum, pacman, homebrew, etc.) since
+they are outside the scope of opam.
+
+opam metadata includes documentation about these external dependencies, on a
+variety of systems/distributions, in the form of a
+[`depexts:`](manual#opamfield-depexts)
+field. Opam should print the required system dependencies, as documented for
+your OS, upon failure, and the `depext` opam plugin can take care of installing
+them for you:
+
+```
+opam depext <opam-packages>
+```
+
+This should install `opam-depext` if needed and prompt to install the system
+packages required by your opam packages or their dependencies, through your OS's
+packaging system.
+
+If that doesn't work...
+* Check for hints printed by the failing package
+* Lookup the development packages corresponding to the error in your system's
+  package repositories.
+* Dependencies for your specific system may not be known, but check the output
+  of `opam list --rec --resolve <package>,<package>... --columns name,depexts:`:
+  it will list dependencies on all known systems and may get you in the right
+  direction.
+
+In any of these cases, that's useful information that was missing from the opam
+repository: we would really appreciate that you take a minute to save others the
+trouble of looking by filling an issue in
+[the opam-repository tracker](https://github.com/ocaml/opam-repository/issues),
+with your system details, the output of `opam config report`, and the solution,
+if you found it. Thanks!
+
+---
+
+#### 🐫  I have weird checksum errors: where do they come from?
+
+First of all, you should make sure your repositories are up-to-date:
+
+```
+opam update
+```
+
+If this isn't enough, or if you get the checksum errors while running `opam
+init`, this could be caused by a badly configured proxy cache that is serving
+stale files. To clear your proxy cache, you can use `wget --no-cache
+<remote-file>` and retry.
+
+As a last resort, you can bypass the checksum checks using `--no-checksums`.
+
+---
+
+#### 🐫  opam is prompting me to install or upgrade packages that I am not interested in, or doesn't install the latest version by default. Why? What can I do?
+
+* You can be more explicit in your request (`opam upgrade PACKAGES`, `opam
+  install 'PACKAGE>=VERSION' PACKAGE...`, etc.). The latest version may not be
+  available on your system, in this case this will tell you why.
```

Comment:
```suggestion
  install 'PACKAGE>=VERSION' PACKAGE...`, etc.). If the latest version is not
  available on your system, this will tell you why.
```